src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,23 +26,14 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> messages;
-
-  public void init(){
-      messages = new ArrayList<>();
-      messages.add("Attack on Titan");
-      messages.add("Naruto");
-      messages.add("Bleach");
-      messages.add("Demon Slayer");
-      messages.add("Death Note");
-      messages.add("Dragon Ball Z/Super");
-      messages.add("Yu-Gi-Oh");
-  }
+  private List<String> messages = new ArrayList<>();
 
-/** Converts message of arrays into json fromat **/
-  private String convertToJsonUsingGson(){
-      String json = new Gson().toJson(messages);
-      return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-input");
```

Comment:
Move string "comment-input" to static const. The logic here is that you might decide to change the parameter name in future and this might break the server logic. Future Erick might have a tough time figuring out why the server is not behaving as expected. Having these values as static consts would help in catching these bugs easily and fix them.